src/mapeo-project.js

```diff
@@ -840,6 +856,32 @@ async function deleteAll(dataType) {
   return Promise.all(deletions)
 }
 
+/**
+ * @param {Object} opts
+ * @param {Logger} opts.logger
+ * @param {MapeoProject['translation']} opts.translation
+ * @param {MapeoProject['preset']} opts.presets
+ * @param {MapeoProject['field']} opts.fields
+ */
+async function deleteTranslations(opts) {
+  const translations = await opts.translation.getMany()
+  await Promise.all(
+    translations.map(async ({ docId, docIdRef, schemaNameRef }) => {
+      try {
+        if (schemaNameRef === 'presets' || schemaNameRef === 'fields') {
+          const toDelete = await opts[schemaNameRef].getByDocId(docIdRef)
+          if (toDelete.deleted) opts.translation.delete(docId)
+        }
+      } catch (e) {
+        opts.logger.log(
+          `referred ${schemaNameRef} is not found, deleting translation`
+        )
+        opts.translation.delete(docId)
+      }
```

Comment:
This code has two small problems:

1. We don't `await` the result of `opts.translation.delete()`.
2. If the first `opts.translation.delete()` rejects, we'll just try to delete it again.

Proposed fix:

```suggestion
      if (schemaNameRef !== 'presets' && schemaNameRef !== 'fields') return

      let shouldDelete = true
      try {
        const toDelete = await opts[schemaNameRef].getByDocId(docIdRef)
        shouldDelete = toDelete.deleted
      } catch (e) {
        opts.logger.log(
          `referred ${schemaNameRef} is not found, deleting translation`
        )
      }

      if (shouldDelete) {
        opts.translation.delete(docId)
      }
```

Reply:
this makes sense. I think I was trying to catch to cases on this:
1. Delete every `translation` linked to by a `preset` or `field` (since we are also deleting those)
2. If there's a translation that links to an unexisting `preset` or `field` we should also deleted (since its basically a _useless_ translation)

Reply:
Additionally, typescript doesn't seem to get that the inverted logic `( !== 'presets' || !== 'fields')` is a typeguard?